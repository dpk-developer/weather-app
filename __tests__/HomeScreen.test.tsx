import React from 'react';
import { render } from '@testing-library/react-native';

import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

import themeReducer, { ThemeState } from '../src/store/slices/themeSlice';
import weatherReducer, { WeatherState } from '../src/store/slices/weatherSlice';

jest.mock('../src/components/SearchBar.tsx', () => () => <></>);
jest.mock('../src/components/WeatherCard.tsx', () => () => <></>);

describe('HomeScreen', () => {
  it('renders correctly with initial state', () => {
    type RootState = {
      weather: WeatherState;
      theme: ThemeState;
    };

    const store = configureStore<RootState>({
      reducer: {
        weather: weatherReducer,
        theme: themeReducer,
      },
      preloadedState: {
        weather: {
          city: 'Paris',
          data: null,
          loading: false,
          error: null,
        },
        theme: { mode: 'light' },
      },
    });

    const HomeScreen = require('../src/screens/Homescreen').default;

    const { toJSON } = render(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
