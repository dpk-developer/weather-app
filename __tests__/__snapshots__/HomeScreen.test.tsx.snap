// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly with initial state 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
      }
    }
  >
    Loading UI...
  </Text>
</RCTSafeAreaView>
`;
